### PR TITLE
[Naitei] Search_Product_Admin

### PR DESCRIPTION
**Pull request by ThaiKhacDuoc**
## Related Tickets
- ticket redmine

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
https://github.com/user-attachments/assets/d0a04846-88e0-4cbd-9d01-515db20c0d06

## Notes (Kiế

n thức tìm hiểu thêm)

### CODE REVIEW

**Comment by ThaiKhacDuoc** [comment]
ready

File: app/views/devise/passwords/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<h2>Change your password</h2>
+
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+  <%= f.hidden_field :reset_password_token %>
+
+  <div class="field">
+    <%= f.label :password, "New password" %><br />
```
Comment (ghost) [inline app/views/devise/passwords/edit.html.erb]: i18n

File: app/views/devise/passwords/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Send me reset password instructions" %>
```
Comment (ghost) [inline app/views/devise/passwords/new.html.erb]: i18n

**Review by ghost — COMMENTED** [review]
có một số chỗ chưa i18n em check lại nhé

**Comment by ThaiKhacDuoc** [comment]
ready

File: app/controllers/admin/products_controller.rb [hunk 3]
```diff
@@ -3,7 +3,10 @@ class Admin::ProductsController < AdminController
   include Pagy::Backend
 
   def index
-    @pagy, @products = pagy(Product.newest, items: Settings.page_size)
+    @q = Product.ransack(search_params)
```
Comment (ghost) [inline app/controllers/admin/products_controller.rb]: bỏ ()

File: app/controllers/admin/products_controller.rb [hunk 4]
```diff
@@ -55,4 +58,23 @@ def find_product
   def product_params
     params.require(:product).permit(Product::PERMITTED_ATTRIBUTES)
   end
+
+  def search_params
+    q_params = params[:q] || {}
+    q_params[:price_gteq] = sanitize_price(q_params[:price_gteq])
```
Comment (ghost) [inline app/controllers/admin/products_controller.rb]: bỏ bớt () đi em

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ThaiKhacDuoc** [comment]
ready

**Review by ghost — APPROVED** [review]

**Pull request merged**